From eee8ec9e8e62e0996dc81e53811005ce67af76f4 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Fri, 30 Aug 2013 15:29:10 +0800
Subject: [PATCH 0399/1543] ENGR00277663-3 power: sabresd_battery: remove
 check charger offset on mx6slevk

commit c39f1821e423e1d6c8689c42dbf0bbef893d2881 from
git://git.freescale.com/imx/linux-2.6-imx.git

No need check the charger offset on mx6slevk, since there is no adc converter

Signed-off-by: Robin Gong <b38343@freescale.com>
(cherry picked from commit 7683c62cf2290629f09983744d5a3987a4d64669)
---
 drivers/power/sabresd_battery.c |   26 ++++++++++++++------------
 1 files changed, 14 insertions(+), 12 deletions(-)

diff --git a/drivers/power/sabresd_battery.c b/drivers/power/sabresd_battery.c
index 7949890..0acd16c 100644
--- a/drivers/power/sabresd_battery.c
+++ b/drivers/power/sabresd_battery.c
@@ -623,18 +623,20 @@ static struct max8903_pdata *max8903_of_populate_pdata(
 		dev_err(dev, "pin pdata->flt: invalid gpio %d\n", pdata->flt);
 		return NULL;
 	}
-
-	if (of_property_read_u32(of_node, "offset-charger",
-		&offset_charger))
-		dev_err(dev, "Not setting offset-charger in dts!\n");
-
-	if (of_property_read_u32(of_node, "offset-discharger",
-		&offset_discharger))
-		dev_err(dev, "Not setting offset-discharger in dts!\n");
-
-	if (of_property_read_u32(of_node, "offset-usb-charger",
-		&offset_usb_charger))
-		dev_err(dev, "Not setting offset-usb-charger in dts!\n");
+	/* no need check offset without adc converter */
+	if (!pdata->feature_flag) {
+		if (of_property_read_u32(of_node, "offset-charger",
+			&offset_charger))
+			dev_err(dev, "Not setting offset-charger in dts!\n");
+
+		if (of_property_read_u32(of_node, "offset-discharger",
+			&offset_discharger))
+			dev_err(dev, "Not setting offset-discharger in dts!\n");
+
+		if (of_property_read_u32(of_node, "offset-usb-charger",
+			&offset_usb_charger))
+			dev_err(dev, "Not setting offset-usb-charger in dts!\n");
+	}
 
 	return pdata;
 }
-- 
1.7.5.4

